### PR TITLE
Fix: Gracefully handle invalid browserslist and URL queries

### LINKED ISSUE #1666: [extension-browser] Specifying an invalid browsers query hangs analysis [1]
Found by @ststimac 

Actual:
> a user can enter an invalid query or resource and start the scan. The webhint tool appears to be running as it hits the screen where the tool says it is analyzing the website that is open. After a few minutes nothing appears in the tool and it stills shows that it is analyzing.  

Expected:
> If a user mistypes or enters an invalid browser target query or a third party resource there should be some indication from the tool that the custom query or resource is invalid and the scan cannot be performed. 

I dug into this a bit and specifically it's our `browserslist` dependency that's rejecting the input. While we could (and should) handle the error, this happens rather late in the process as part of the analysis script injected into a page. As such, we may want to also validate the query ahead of time to provide more timely feedback.

### PR DESCRIPTION
**Pull request by antross**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- ~Added/Updated related documentation.~
- [x] Added/Updated related tests.

## Short description of the change(s)

Fix #1666

Prevent form submission when analysis is run with an invalid
`browserslist` query. Similarly, prevent form submission
when an invalid `RegExp` is provided for ignored URLs.

Update the analysis script gracefully handle invalid should
any be successfully submitted by falling back to defaults.

Also provide feedback to the user during query construction on
whether or not the query is valid and why.

### CODE REVIEW

**Comment by antross** [comment]
Extension bundle with the fix for testing:
* [webhint-browser-extension-fix-1666.zip](https://github.com/webhintio/hint/files/2757424/webhint-browser-extension-fix-1666.zip)

File: packages/extension-browser/src/devtools/panel/views/pages/configuration.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import escapeRegExp = require('lodash/escapeRegExp');
+import browserslist = require('browserslist'); // `require` used because `browserslist` exports a function.
```
Comment (molant) [inline packages/extension-browser/src/devtools/panel/views/pages/configuration.ts]: Don't think we need this comment, the line above also uses `require`

File: packages/extension-browser/src/devtools/panel/views/pages/configuration.ts [hunk 2]
```diff
@@ -61,6 +67,24 @@ export default function view({ onAnalyzeClick }: Props) {
...
+
+        try {
+            browserslist(getBrowsersList());
+            input.setCustomValidity('');
+        } catch (e) {
+            // Report errors, stripping messages about "old" browserslist since the user won't have control over that.
+            input.setCustomValidity(e.message.replace(' Maybe you are using old Browserslist or made typo in query.', ''));
+        }
+
+        // Wait a beat before reporting validity to make Chrome happy.
```
Comment (molant) [inline packages/extension-browser/src/devtools/panel/views/pages/configuration.ts]:
```suggestion
        // Wait a bit before reporting validity to make Chrome happy.
```

**Comment by molant** [comment]
[I've created an issue for the checkboxes vs radios](https://github.com/webhintio/hint/issues/1685)

**Comment by antross** [comment]
@molant This is updated with your feedback, including our offline discussion about preventing the form submission if any input is invalid.

**Pull request merged**